src/script/page/RightSidebar/ConversationDetails/ConversationDetailsHeader.tsx

```diff
@@ -0,0 +1,157 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {FC, ChangeEvent, KeyboardEvent, useEffect, useRef, useState} from 'react';
+import {t} from 'Util/LocalizerUtil';
+import Icon from 'Components/Icon';
+import {ConversationRepository} from '../../../conversation/ConversationRepository';
+import GroupDetails from './GroupDetails';
+import {isEnterKey} from 'Util/KeyboardUtil';
+import {removeLineBreaks} from 'Util/StringUtil';
+import {User} from '../../../entity/User';
+import {ServiceEntity} from '../../../integration/ServiceEntity';
+
+interface ConversationDetailsHeaderProps {
+  isActiveGroupParticipant: boolean;
+  canRenameGroup: boolean;
+  displayName: string;
+  updateConversationName: (conversationName: string) => void;
+  isGroup: boolean;
+  userParticipants: User[];
+  serviceParticipants: ServiceEntity[];
+  allUsersCount: number;
+  isTeam?: boolean;
+}
+
+const ConversationDetailsHeader: FC<ConversationDetailsHeaderProps> = ({
+  isActiveGroupParticipant,
+  canRenameGroup,
+  displayName,
+  updateConversationName,
+  isGroup,
+  userParticipants,
+  serviceParticipants,
+  allUsersCount,
+  isTeam = false,
+}) => {
+  const textAreaRef = useRef<HTMLTextAreaElement | null>(null);
+
+  const [isEditingName, setIsEditingName] = useState<boolean>(false);
+  const [groupName, setGroupName] = useState(displayName);
+
+  const clickToEditGroupName = () => {
+    if (isActiveGroupParticipant) {
+      setIsEditingName(true);
+    }
+  };
+
+  const renameConversation = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    const {value: currentValue} = event.target;
+    setGroupName(currentValue);
+  };
+
+  const handleRenameConversation = (event: KeyboardEvent<HTMLTextAreaElement>) => {
+    if (isEnterKey(event)) {
+      event.preventDefault();
+      const {value: currentValue} = event.currentTarget;
+
+      const currentConversationName = displayName.trim();
+      const newConversationName = removeLineBreaks(currentValue.trim());
+
+      const isNameChanged = newConversationName !== currentConversationName;
+
+      if (isNameChanged) {
+        updateConversationName(newConversationName);
+        setGroupName(newConversationName);
+        setIsEditingName(false);
+      }
+    }
+  };
+
+  // TODO: Update height of textarea on change
+  useEffect(() => {
+    if (isEditingName) {
+      if (textAreaRef.current) {
+        const {scrollHeight} = textAreaRef.current;
+        textAreaRef.current.style.height = `${scrollHeight}px`;
+      }
+
+      setTimeout(() => {
+        const currentValue = textAreaRef.current?.value;
+        const caretPosition = currentValue?.length || 0;
+
+        textAreaRef.current?.setSelectionRange(caretPosition, caretPosition);
+        textAreaRef.current?.focus();
+      }, 0);
+    }
+  }, [textAreaRef, isEditingName, groupName]);
```

Comment:
```suggestion
  }, [isEditingName, groupName]);
```

Ref cannot be put to dependencies, they won't trigger an effect. If you need an effect triggering with the ref change, you need to use `useEffectRef`

Reply:
useEffectRef not needed ;)